platform/src/main/python/dlpx/virtualization/platform/_linked.py

```diff
@@ -719,3 +781,62 @@ def to_protobuf_ownership_spec(ownership_spec):
                 ownership_spec)
 
         return staged_mount_spec_response
+
+    def _internal_staged_source_size(self, request):
+        """Staged Source Size Wrapper for staged plugins.
+
+        Executed as part of several operations to get the source size
+        of a staged source
+
+        Run source_size operation for a staged source.
+
+        Args:
+           request (StagedSourceSizeRequest): Source Size Request arguments.
+
+        Returns:
+           StagedSourceSizeResponse: A response containing the return value -
+           StagedSourceSizeResult which has source size. In
+           case of errors, response object will contain PluginErrorResult.
+        """
+        # Reasoning for method imports are in this file's docstring.
+        from generated.definitions import RepositoryDefinition
+        from generated.definitions import LinkedSourceDefinition
+        from generated.definitions import SourceConfigDefinition
+
+        #
+        # While linked.source_size() is not a required operation, this should
+        # not be called if it wasn't implemented.
+        #
+        if not self.source_size_impl:
+            raise OperationNotDefinedError(Op.LINKED_SOURCE_SIZE)
+
+        staged_source_definition = LinkedSourceDefinition.from_dict(
+            json.loads(request.staged_source.linked_source.parameters.json))
+        mount = Mount(
+            remote_environment=(RemoteEnvironment.from_proto(
+                request.staged_source.staged_mount.remote_environment)),
+            mount_path=request.staged_source.staged_mount.mount_path,
+            shared_path=request.staged_source.staged_mount.shared_path)
+        staged_source = StagedSource(
+            guid=request.staged_source.linked_source.guid,
+            source_connection=RemoteConnection.from_proto(
+                request.staged_source.source_connection),
+            parameters=staged_source_definition,
+            mount=mount,
+            staged_connection=RemoteConnection.from_proto(
+                request.staged_source.staged_connection))
+
+        repository = RepositoryDefinition.from_dict(
+            json.loads(request.repository.parameters.json))
+        source_config = SourceConfigDefinition.from_dict(
+            json.loads(request.source_config.parameters.json))
+
+        size = self.source_size_impl(
```

Comment:
nit: same comment here and other places - size -> source_size

Reply:
Sure, will be taking care in separate JIRA.